doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,614 @@
+# Appending vs Overwriting Files
+
+## Introduction
+
+Pachyderm is designed to work with pipelined data processing in a
+containerized environment.  The Pachyderm File System (pfs) is a
+file-based system that is distributed and supports data of all types
+of files (binary, csv, json, images, etc) from many sources and
+users. That data is processed in parallel across many different jobs
+and pipeline using the Pachyderm Pipeline System (pps). The Pachyderm
+File System (pfs) and Pachyderm Pipeline System (pps) are designed to
+work together to get the right version of the right data to the right
+container at the right time.
+
+Among the many complexities you must consider are these:
+
+- Files can be put into pfs in "append" or "overwrite" mode.
+- Pipeline definitions use "glob" operators to filter a view of input repositories.
+- Output repositories must merge data from what may be multiple containers running the same pipeline code, at the same time.
+
+When you add in the ability to do "cross" and "union" operators on
+multiple input repositories to those three considerations, it can be a
+little confusing to understand what's actually happening with your
+files!
+
+This document will take you through loading data into pachyderm,
+consuming that data in input repos and placing data into output
+repositories to show you the various options and conventions of
+working with data in Pachyderm.
+
+## Loading data into Pachyderm
+
+### Overwriting files
+
+When putting files into a pfs repo via Pachyderm's `pachctl` utility,
```

Comment:
Slight tweak, you may want to specify that these are just general api features rather than something specific to `pachctl`.

Reply:
Good point, clarified and recommitted!